src/libsystemd-network/dhcp6-option.c

```diff
@@ -564,13 +580,18 @@ int dhcp6_option_parse_ia(sd_dhcp6_client *client, DHCP6Option *iaoption, DHCP6I
                 if (len < DHCP6_OPTION_IA_TA_LEN)
                         return -ENOBUFS;
 
+                /* According to RFC8415, IAs which do not match the client's IAID should be ignored,
+                 * but not necessary to ignore or refuse the whole message. */
+                if (((const struct ia_ta*) iaoption->data)->id != iaid)
+                        return -ENOMSG; /* Indicate the option should be ignored. */
```

Comment:
Maybe this deserves a debug message also, so that people with misconfigured environments get some feedback like before.

Reply:
Thank you for the suggestion. A debug message is added.

Reply:
Thanks :)